### PR TITLE
windows: fix bun plugin

### PR DESCRIPTION
**Pull request by paperclover**
we no longer treat filename "C:\\hello" as namespcae C file path \hello. this change only impacts windows

### CODE REVIEW

File: test/js/bun/plugin/muh.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { plugin } from "bun";
```
Comment (Jarred-Sumner) [inline test/js/bun/plugin/muh.ts:1]: Where is `muh.ts` loaded?

File: src/js/builtins/BundlerPlugin.ts [hunk 2]
```diff
@@ -253,7 +253,9 @@ export function runOnResolvePlugins(this: BundlerPlugin, specifier, inputNamespa
                 throw new TypeError('onResolve plugin "path" must be absolute when the namespace is "file"');
               }
             } else {
-              // TODO: Windows
+              if (require("path").isAbsolute(path) === false || path.includes("..")) {
```
Comment (Jarred-Sumner) [inline src/js/builtins/BundlerPlugin.ts]:
```suggestion
              if (require("node:path").isAbsolute(path) === false || path.includes("..")) {
```

**Comment by Jarred-Sumner** [comment]
what about UNC paths

File: test/js/bun/plugin/muh.ts (hunk 1 above)
Comment (paperclover) [reply test/js/bun/plugin/muh.ts:1]: testing garbage file to be deleted

**Comment by paperclover** [comment]
> what about UNC paths

a unc path does not have a colon

\\\\server\share\path

**Pull request merged**